File: Cpp/690_Employee_Importance.cpp
/*
// Definition for Employee.
class Employee {
public:
    int id;
    int importance;
    vector<int> subordinates;
};
*/

class Solution {
public:
    int getImportance(vector<Employee*> employees, int id) {
        int result = 0;
        for(auto& employee : employees){
            if(employee->id!=id) continue;
            result += employee->importance;
            for(auto& subordinate : employee->subordinates){
                result += getImportance(employees, subordinate);
            }
            return result;
        }
        
        return result;
    }
};

File: Kotlin/690_Employee_Importance.kt
/*
 *	// Definition for Employee.
 *	class Employee {
 *		var id:Int = 0
 *		var importance:Int = 0
 *		var subordinates:List<Int> = listOf()
 *	}
 */

class Solution {
    fun getImportance(employees: List<Employee?>, id: Int): Int {
        var result = 0
        
        employees.forEach{ employee ->
            if(employee?.id == id){
                result += employee.importance
                if (employee.subordinates.isEmpty()) return result
                
                employee.subordinates.forEach{ subordinate ->
                    result += getImportance(employees, subordinate)
                }
                
                return result
            }
        }
        
        return result
    }
}